packages/solana-wallet-injection/src/wallet.ts

```diff
@@ -0,0 +1,184 @@
+/* eslint-disable no-console */
+import { Liquality, LiqualityEvent } from '@liquality/solana-wallet-standard/src/window';
+import {
+  PublicKey,
+  Transaction,
+  VersionedTransaction,
+  SendOptions,
+  Connection,
+  Message,
+  Keypair,
+} from '@solana/web3.js';
+import { SolanaParser, ParsedInstruction } from '@debridge-finance/solana-transaction-parser';
+import { Idl } from '@project-serum/anchor';
+import { COMMON_REQUEST_MAP } from './types';
+import { BigNumber } from '@chainify/types';
+import base58 from 'bs58';
+
+/**
+ * @dev users are paying a fixed fee of 5000 lamports per transaction/signature
+ */
+const DEFAULT_FEE = '0.000005';
+
+export class LiqualitySolanaWallet implements Liquality {
+  public publicKey: PublicKey | null;
+  public isConnected: boolean;
+  private window: Window;
+  private _connection: Connection;
+
+  public get connection(): Connection {
+    return this._connection;
+  }
+  public set connection(value: Connection) {
+    this._connection = value;
+  }
+  private solanaParser = new SolanaParser([]);
+
+  constructor(window: Window) {
+    this.window = window;
+    // TODO: take rpc url from cryptoassets
+    this.connection = new Connection(
+      'https://red-sleek-rain.solana-mainnet.discover.quiknode.pro/fc112deb1e0228d09d0d8c12b8de5a601d251d80'
+    );
+  }
+
+  public async connect(
+    _options?: { onlyIfTrusted?: boolean | undefined } | undefined
+  ): Promise<{ publicKey: PublicKey }> {
+    const { accepted } = await this.enable();
+
+    if (!accepted) {
+      throw new Error('User rejected');
+    }
+
+    const addresses = await this.callMethod(COMMON_REQUEST_MAP.wallet_getAddresses);
+    const [{ publicKey }] = addresses;
+    this.publicKey = new PublicKey(publicKey);
+    this.isConnected = true;
+
+    console.debug(`Public Key`, this.publicKey);
+    console.debug(`Public Key Decoded`, this.publicKey.toBase58());
+    console.log('Is Connected: ', this.isConnected);
+    return { publicKey };
+  }
+
+  public async disconnect(): Promise<void> {
+    this.publicKey = null;
+    this.isConnected = false;
+  }
+
+  public async signAndSendTransaction<T extends Transaction | VersionedTransaction>(
+    transaction: T,
+    options?: SendOptions | undefined
+  ): Promise<{ signature: string }> {
+    console.debug('signAndSendTransaction');
```

Comment:
@kraikov Is there an intermittent bug you're still trying to track down by these console statements?

Reply:
usually injection is very problematic from my experience with the rest of the injection implemnetations

I left these so they can be easily debugged by QA